src/idl/src/scope.c

```diff
@@ -452,7 +452,10 @@ idl_resolve(
   if (pstate->flags & IDL_FLAG_CASE_SENSITIVE)
     ignore_case = 0u;
 
-  scope = (scoped_name->absolute) ? pstate->global_scope : pstate->scope;
+  if (pstate->parser.state == IDL_PARSE_ANNOTATION_APPL_PARAMS)
+    scope = pstate->annotation_scope;
+  else
+    scope = (scoped_name->absolute) ? pstate->global_scope : pstate->scope;
```

Comment:
I'm not sure this is correct. It reads like the current scope isn't searched if something can't be found in the annotation declaration scope. I'm also not sure what the spec says about this... Should the scenario below work?

```
module m1 {
  @annotation a1 {
    unsigned int32 i1;
  };
  module m2 {
    const unsigned int32 i2 = 1;
    @a1(i1 = i2) struct s1 {
      char c1;
    };
  };
};
```

Reply:
Good question, I'm not sure if it should and I can't really find the answer in the spec. The specs states that enums that are defined in the annotation body may be used unscoped, which could imply that a scoped name should be used for const expressions from outside the annotation body scope (i.e. `i1 = m2::i2') and that works with the changes I've added in commit 7b1f554. This commit also adds an additional test for annotation parameter scope.